js/dragonex.js

```diff
@@ -0,0 +1,912 @@
+        let cost = amount * rate;
+        let precision = market['precision']['price'];
+        if (side === 'sell') {
+            cost *= price;
+        } else {
+            key = 'base';
+            precision = market['precision']['amount'];
+        }
+        cost = this.decimalToPrecision (cost, ROUND, precision, this.precisionMode);
+        return {
+            'type': takerOrMaker,
+            'currency': market[key],
+            'rate': rate,
+            'cost': parseFloat (cost),
+        };
+    }
+
+    async fetchBalance (params = {}) {
+        await this.loadMarkets ();
+        const response = await this.privatePostUserOwn (params);
+        const result = { 'info': response };
+        const balances = this.safeValue (response, 'data', []);
+        for (let i = 0; i < balances.length; i++) {
+            const balance = balances[i];
+            const currencyId = balance['coin_id'];
+            const code = this.safeCurrencyCode (currencyId);
+            const account = this.account ();
+            account['free'] = this.safeFloat (balance, 'volume') - this.safeFloat (balance, 'frozen');
+            account['used'] = this.safeFloat (balance, 'frozen');
+            account['total'] = this.safeFloat (balance, 'volume');
+            result[code] = account;
+        }
+        return this.parseBalance (result);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol_id': market['symbol_id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = limit; // default = maximum = 100
+        }
+        const response = await this.v1GetMarketDepth (this.extend (request, params));
+        const orderBook = this.parseOrderBook (response);
+        const buysList = [];
+        const sellsList = [];
+        for (let i = 0; i < this.safeValue (response, 'data', {})['buys'].length; i++) {
+            const buys = [];
+            buys.push (this.safeInteger (this.safeValue (response, 'data', {})['buys'][i], 'price', 0));
+            buys.push (this.safeInteger (this.safeValue (response, 'data', {})['buys'][i], 'volume', 0));
+            buysList.push (buys);
+        }
+        for (let i = 0; i < this.safeValue (response, 'data', {})['sells'].length; i++) {
+            const sells = [];
+            sells.push (this.safeInteger (this.safeValue (response, 'data', {})['sells'][i], 'price', 0));
+            sells.push (this.safeInteger (this.safeValue (response, 'data', {})['sells'][i], 'volume', 0));
+            sellsList.push (sells);
+        }
+        orderBook['bids'] = buysList;
+        orderBook['asks'] = sellsList;
+        return orderBook;
+    }
+
+    parseTicker (ticker, market = undefined) {
+        ticker = this.safeValue (this.safeValue (ticker, 'data', {}), 'list', [])[0];
+        const lastNum = ticker.length - 1;
```

Comment:
This line (and several other similar lines) violate the contributing guidelines:

- https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#working-with-array-lengths